components/governance/org.wso2.carbon.governance.taxonomy/src/main/java/org/wso2/carbon/governance/taxonomy/internal/TaxonomyServiceComponent.java

```diff
@@ -43,15 +63,63 @@
      *
      * @param context OSGi component context.
      */
-    protected void activate(ComponentContext context) {
+    protected void activate(ComponentContext context)
+            throws UserStoreException, RegistryException, ParserConfigurationException, SAXException,
+            XPathExpressionException, IOException {
```

Comment:
You can reduce the complexity by removing the "if (log.isDebugEnabled())" for formatted(no appeders) strings.